lms/djangoapps/api_manager/receivers.py

```diff
@@ -17,3 +18,13 @@ def on_course_deleted(sender, **kwargs):  # pylint: disable=W0613
     course_key = kwargs['course_key']
     CourseGroupRelationship.objects.filter(course_id=course_key).delete()
     CourseContentGroupRelationship.objects.filter(course_id=course_key).delete()
+
+
+@receiver(SignalHandler.course_published)
+def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
```

Comment:
we should add a new listen_for_course_publish() signal handler (ie, receiver) to the new app and move the metadata update to there -- the api_manager should not be responsible for informing the rest of the system about state changes through the signals framework.